interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/Validations.scala

```diff
@@ -56,14 +57,21 @@ object Validations {
   private def validateWithCustomValidators[T >: PartSubGraphCompilationError <: ProcessCompilationError](parameterDefinitions: List[Parameter],
                                                                                                          parameters: List[evaluatedparam.Parameter])
                                                                                                         (implicit nodeId: NodeId) = {
-    val validators = parameterDefinitions.map(param => (param.name, param.validators)).toMap
-    val paramWithValidatorList = for {
+    val definitionsMap = parameterDefinitions.map(param => (param.name, param)).toMap
+    val validationResults = for {
       param <- parameters
-      validator <- validators.getOrElse(param.name, List.empty)
-    } yield (param, validator)
-    val validationResults = paramWithValidatorList.map {
-      case (param, validator) => validator.isValid(param.name, param.expression.expression, None).toValidatedNel
-    }
+      paramDefinition <- definitionsMap.get(param.name)
+      paramValidationResult = validateParameterWithCustomValidators(paramDefinition, param)
+    } yield paramValidationResult
     validationResults.sequence.map(_ => Unit)
   }
+
+  def validateParameterWithCustomValidators[T >: PartSubGraphCompilationError <: ProcessCompilationError](paramDefinition: Parameter,
```

Comment:
I wonder, could it be a method in Parameter? Or do we want to keep them anemic ;) 
Also, I think 'custom' is misleading here - they are in most cases rather standard?

Reply:
fixed